tsl/src/gapfill/planner.c

```diff
@@ -336,8 +425,82 @@ plan_add_gapfill(PlannerInfo *root, RelOptInfo *group_rel)
 
 		foreach (lc, copy)
 		{
-			add_path(group_rel, gapfill_path_create(root, lfirst(lc), context.call));
+			add_path(group_rel, gapfill_path_create(root, lfirst(lc), context.call.func));
 		}
 		list_free(copy);
 	}
 }
+
+static inline bool
+is_gapfill_path(Path *path)
+{
+	return IsA(path, CustomPath) && castNode(CustomPath, path)->methods == &gapfill_path_methods;
+}
+
+/*
+ * Since we construct the targetlist for the gapfill node from the
+ * final targetlist we need to adjust any intermediate targetlists
```

Comment:
maybe better wording: the final targetlist of the top-level windowagg node. We need to adjust any child winodowagg node?

But again I think a picture of the hierarchy would really help here